packages/ilib-loctool-mrkdwn/MrkdwnJsFile.js

```diff
@@ -809,10 +809,11 @@ MrkdwnJsFile.prototype.localizeText = function(translations, locale) {
     }
 
     var output;
-    if (!this.project.settings || this.project.settings.outputStyle === "commonjs") {
-        output = "module.exports.messages = ";
-    } else {
+    if (!this.project.settings || !this.project.settings.mrkdwn ||
+         this.project.settings.mrkdwn.outputStyle !== "commonjs") {
         output = "export default messages = ";
+    } else {
+        output = "module.exports.messages = ";
     }
```

Comment:
```suggestion
     if (this.project.settings?.mrkdwn?.outputStyle === "commonjs") {
           output = "module.exports.messages = ";
    } else {
        output = "export default messages = ";
    }
```

Reply:
I'd revers the `if` for better readability. Or at least changed 
```
if (!this.project.settings || !this.project.settings.mrkdwn || this.project.settings.mrkdwn.outputStyle !== "commonjs") { ... }
```
to 
```
if (this.project.settings?.mrkdwn?.outputStyle !== "commonjs") { ... }
```

Reply:
Source: [Optional chaining (?.)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

Reply:
optional chaining is also an ESM feature